Weather App project

SofiBretz opened the pull request:
**In this project I:**

- Add the 'Weather API' and configure it with the logic of the app.
- Add the 'Webpack' and 'eslint' packages.
- Add style to the app.
- Add the readme file with the image example and the 'demo' link.
- Program the logic of the app.
- Separate the main module form the DOM.

shloch commented on README.md:
```diff
@@ -1 +1,66 @@
+
+- You have to get a text editor to inspect the code in your computer.
+
+### Install
+
+- Install a text editor (if you don't have already).
+- Clone the project to your computer.
+
```
specify the commands to install and setup the project

SofiBretz replied:
Thanks for this.

shloch commented on .eslintrc.json:
```diff
@@ -0,0 +1,19 @@
+{
+    "env": {
+        "browser": true,
+        "es6": true
+    },
+    "extends": [
+        "airbnb-base"
+    ],
+    "globals": {
+        "Atomics": "readonly",
+        "SharedArrayBuffer": "readonly"
+    },
+    "parserOptions": {
+        "ecmaVersion": 2018,
+        "sourceType": "module"
+    },
+    "rules": {
+    }
+}
```
use the same linting config file provided by microverse ([ click here to access file](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json) )

SofiBretz replied:
Done, thanks!

shloch commented on src/index.js:
```diff
@@ -0,0 +1,108 @@
+    });
+}
+
+function getWeather(location) {
+  const api = `http://api.openweathermap.org/data/2.5/weather?q=${location}&APPID=${key}`;
+
+  fetch(api)
+    .then((response) => {
+      const data = response.json();
+      return data;
+    })
+    .then((data) => {
+      weather.temperature.value = Math.floor(data.main.temp - KELVIN);
+      weather.description = data.weather[0].description;
+      weather.iconId = data.weather[0].icon;
+      weather.city = data.name;
+      weather.country = data.sys.country;
+    })
+    .then(() => {
+      displayWeatherInfo(iconElement, tempElement, descElement, locationElement, weather);
+    });
+}
```
### Great work with the API...this is great

SofiBretz replied:
Thank you!

shloch commented on src/index.js:
```diff
@@ -0,0 +1,108 @@
+  getWeatherAuto(latitude, longitude);
+}
+
+function showError(error) {
+  notificationElement.style.display = 'block';
+  notificationElement.innerHTML = `<p> ${error.message} </p>`;
+}
+
```
Look for a way to separate LOGICAL code from DOM. This file manipulates the API and DOM elements. You could create a file for Weather Api that does calls and pass data to another DOM file, where you do all the displays [ [see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) ]

SofiBretz replied:
Done, thank you.

SofiBretz commented:
**Summary of Changes:**

- Added the 'Microverse eslint' configuration version.
- Added the install commands on the readme file.
- Created a new function to handle DOM for the error and for the temperature display.
- ESlint fixed, no offenses detected.

Ready for a code review, thanks! :)
